django/views/generic/detail.py

```diff
@@ -100,6 +100,11 @@ def get_context_data(self, **kwargs):
         Insert the single object into the context dict.
         """
         context = {}
+        # when self.object value is None, we should NOT run get_object()
+        # method; this value means object will be created by ModelForm (in
+        # CreateView)
+        if not hasattr(self, "object"):
```

Comment:
FWIW, this couple of lines was previously proposed in https://github.com/django/django/pull/2260.
@mjtamlyn, since you said you didn't like that other change, maybe you could say if you like this one instead?